aws/rust-runtime/aws-types/src/credentials/provider.rs

```diff
@@ -138,22 +143,29 @@ impl Error for CredentialsError {
     }
 }
 
+/// Result type for credential providers.
 pub type Result = std::result::Result<Credentials, CredentialsError>;
 
+/// Convenience `ProvideCredentials` struct that implements the `ProvideCredentials` trait.
 pub mod future {
     use aws_smithy_async::future::now_or_later::NowOrLater;
     use std::future::Future;
     use std::pin::Pin;
     use std::task::{Context, Poll};
 
     type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
+
+    /// Convenience `ProvideCredentials` struct that implements the `ProvideCredentials` trait.
+    #[derive(Debug)]
```

Comment:
this isn't right—this is the future newtype that `ProvideCredentials` must return